File: sdks/typescript/tests/signal-catalogue.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import {
  OPENAPI_DOCUMENT_SHA256,
  SIGNAL_CATALOGUE,
  SIGNAL_CATALOGUE_JSON,
  SIGNAL_CATALOGUE_SHA256,
} from "../src/index.js";

const EXPECTED_OPENAPI_SHA256 = "8f9efc7be194e4900812f93b422e252fab187facf854c9459f1c84be70971f8b";
const EXPECTED_SIGNAL_SHA256 = "239df2ced5974d4351ca01565369646964cb65c63d0fbae01dddccb4c3ac5a07";

test("generated signal catalogue is public, exact, and inert", () => {
  const parsed = JSON.parse(SIGNAL_CATALOGUE_JSON) as {
    contract_version: number;
    diagnostic_levels: unknown[];
    metric_attributes: unknown[];
    metric_instruments: unknown[];
  };

  assert.equal(SIGNAL_CATALOGUE_SHA256, EXPECTED_SIGNAL_SHA256);
  assert.equal(OPENAPI_DOCUMENT_SHA256, EXPECTED_OPENAPI_SHA256);
  assert.deepEqual(SIGNAL_CATALOGUE, parsed);
  assert.equal(parsed.contract_version, 1);
  assert.equal(parsed.diagnostic_levels.length, 4);
  assert.equal(parsed.metric_attributes.length, 9);
  assert.equal(parsed.metric_instruments.length, 22);
  assert.equal(Object.isFrozen(SIGNAL_CATALOGUE), true);
  assert.equal(Object.isFrozen(SIGNAL_CATALOGUE.metric_instruments), true);
  assert.equal(Object.isFrozen(SIGNAL_CATALOGUE.metric_instruments[0]), true);
  assert.doesNotMatch(
    SIGNAL_CATALOGUE_JSON,
    /"(?:activation|active_level|current_level|diagnostic_snapshot|enabled|events|exporter|lifecycle|samples|subscriber|timestamp)"\s*:/,
  );
});

File: sdks/java/src/test/java/io/rrflow/rrd/SignalCatalogueTest.java
package io.rrflow.rrd;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

import org.junit.jupiter.api.Test;
import tools.jackson.databind.JsonNode;
import tools.jackson.databind.json.JsonMapper;

final class SignalCatalogueTest {
    private static final JsonMapper JSON = JsonMapper.builder().build();
    private static final String EXPECTED_OPENAPI_SHA256 =
            "8f9efc7be194e4900812f93b422e252fab187facf854c9459f1c84be70971f8b";
    private static final String EXPECTED_SIGNAL_SHA256 =
            "239df2ced5974d4351ca01565369646964cb65c63d0fbae01dddccb4c3ac5a07";

    @Test
    void generatedSignalCatalogueIsPublicExactAndInert() throws Exception {
        JsonNode catalogue = JSON.readTree(SignalCatalogue.JSON);

        assertEquals(EXPECTED_SIGNAL_SHA256, SignalCatalogue.SIGNAL_CATALOGUE_SHA256);
        assertEquals(EXPECTED_OPENAPI_SHA256, SignalCatalogue.OPENAPI_DOCUMENT_SHA256);
        assertEquals(1, catalogue.path("contract_version").asInt());
        assertEquals(4, catalogue.path("diagnostic_levels").size());
        assertEquals(9, catalogue.path("metric_attributes").size());
        assertEquals(22, catalogue.path("metric_instruments").size());
        for (String forbidden : new String[] {
            "activation", "active_level", "current_level", "diagnostic_snapshot", "enabled",
            "events", "exporter", "lifecycle", "samples", "subscriber", "timestamp"
        }) {
            assertFalse(catalogue.has(forbidden), "signal discovery contains " + forbidden);
        }
    }
}
